packages/compass-data-modeling/src/utils/schema-traversal.tsx

```diff
@@ -165,3 +165,118 @@ export const getFieldFromSchema = ({
     parentFieldPath: [...parentFieldPath, nextInPath],
   });
 };
+
+type MutationParameters = {
+  update: 'removeField' | 'renameField';
+  newFieldName?: string;
+};
+
+const getMutatedSchema = ({
+  schema,
+  fieldName,
+  newFieldName,
+  update,
+}: {
+  schema: MongoDBJSONSchema;
+  fieldName: string;
+} & MutationParameters): MongoDBJSONSchema => {
+  switch (update) {
+    case 'removeField': {
+      if (!schema.properties || !schema.properties[fieldName])
+        throw new Error('Field to remove does not exist');
+      return {
+        ...schema,
+        properties: Object.fromEntries(
+          Object.entries(schema.properties).filter(([key]) => key !== fieldName)
+        ),
+      };
+    }
+    case 'renameField': {
+      if (!schema.properties || !schema.properties[fieldName])
+        throw new Error('Field to rename does not exist');
+      if (!newFieldName)
+        throw new Error('New field name is required for the rename operation');
+      return {
+        ...schema,
+        properties: Object.fromEntries(
+          Object.entries(schema.properties).map(([key, value]) =>
+            key === fieldName ? [newFieldName, value] : [key, value]
+          )
+        ),
+      };
+    }
+    default:
+      return schema;
+  }
+};
+
+/**
+ * Finds a single field in a MongoDB JSON schema and performs an update operation on it.
+ */
+export const updateSchema = ({
```

Comment:
I don't have anything to suggest here immediately, but it bothers me that we added some methods to traverse the schema and are not using them for that purpose adding more methods that all share some similar logic to traverse the schema slightly differently from the traverse method, means we didn't really get the interface we need from the one we added. Should probalby revisit this later